### PR TITLE
[#711] feat(netty): Add Netty port information for Shuffle Server

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add Netty port information for Shuffle Server

### Why are the changes needed?
Let us can use old servers with Grpc and new servers with Netty at the same time.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Add new ut

### CODE REVIEW

File: client-mr/src/main/java/org/apache/hadoop/mapred/RssMapOutputCollector.java [hunk 1]
```diff
@@ -143,20 +143,14 @@ private Map<Integer, List<ShuffleServerInfo>> createAssignmentMap(JobConf jobCon
...
-            serverInfo[0], Integer.parseInt(serverInfo[1]));
-        assignServers.add(sever);
-      }
+      RssMRUtils.buildAssignServers(i, splitServers, assignServers);
       partitionToServers.put(i, assignServers);
     }
     return partitionToServers;
   }
 
+
```
Comment (smallzhongfeng) [inline client-mr/src/main/java/org/apache/hadoop/mapred/RssMapOutputCollector.java:152]: Remove blank line.

**Review by smallzhongfeng — APPROVED** [review]
LGTM except for one minor comment !

**Review by leixm — APPROVED** [review]

**Comment by leixm** [comment]
+1 LGTM.

**Comment by jerqi** [comment]
> I'm not sure whether we should introduce a new port here.. It adds some deployment overhead. Is there ways to multiplex port here?

It's too complex to  support that one port can use  multiple protocols.

**Comment by jerqi** [comment]
> > > > > I'm not sure whether we should introduce a new port here.. It adds some deployment overhead. Is there ways to multiplex port here?
> > > > 
> > > > 
> > > > It's too complex to support that one port can use multiple protocols.
> > > 
> > > 
> > > OK. Then is it still required to use a fixed port for Netty service? I'd like the netty service could be bind for random available port.
> > 
> > 
> > That's another pr. If we support to bind random port, we should support both Grpc and Netty.
> 
> Yeah. That's another PR, but we will support random port, right? Then, I'd like to propose some changes in this PR.

What are changes?

File: common/src/main/java/org/apache/uniffle/common/ClientType.java [hunk 2]
```diff
@@ -18,5 +18,6 @@
 package org.apache.uniffle.common;
 
 public enum ClientType {
-  GRPC
+  GRPC,
+  GRPC_NETTY
```
[the remark this answers is not in the document]
Comment (zuston) [reply common/src/main/java/org/apache/uniffle/common/ClientType.java:22]: I think current implementation is a mixed way, only main data transfer api will be changed to netty proto. So GRPC_NETTY is reasonable.

[the remark this answers is not in the document]
Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/ClientType.java:22]: > I think current implementation is a mixed way, only main data transfer api will be changed to netty proto. So GRPC_NETTY is reasonable.

+ 1.

File: common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java [hunk 3]
```diff
@@ -25,19 +25,28 @@ public class ShuffleServerInfo implements Serializable {
 
   private String host;
 
-  private int port;
+  private int grpcPort;
+
+  private int nettyPort = 0;
```
[the remark this answers is not in the document]
Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/ShuffleServerInfo.java:30]: ok.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/ServerNode.java [hunk 4]
```diff
@@ -40,6 +40,7 @@ public class ServerNode implements Comparable<ServerNode> {
   private boolean isHealthy;
   private final ServerStatus status;
   private Map<String, StorageInfo> storageInfo;
+  private int nettyPort = 0;
```
[the remark this answers is not in the document]
Comment (jerqi) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/ServerNode.java:43]: ok.

File: server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java [hunk 5]
```diff
@@ -397,6 +397,13 @@ public class ShuffleServerConf extends RssBaseConf {
       .defaultValue(true)
       .withDescription("Whether shutdown the server after server is decommissioned");
 
+  public static final ConfigOption<Integer> NETTY_SERVER_PORT = ConfigOptions
+      .key("rss.server.netty.port")
+      .intType()
+      .checkValue(ConfigUtils.POSITIVE_INTEGER_VALIDATOR_2, "netty port must be positive")
+      .defaultValue(0)
```
[the remark this answers is not in the document]
Comment (jerqi) [reply server/src/main/java/org/apache/uniffle/server/ShuffleServerConf.java:404]: ok.

**Review by zuston — APPROVED** [review]

**Pull request merged**